src/cljs/ui/component/maps.cljs

```diff
@@ -0,0 +1,213 @@
+                                      y (-> evt .-nativeEvent .-offsetY)]
+                                  (cond
+
+                                    ;; left button pressed
+                                    (= left-button (.-button evt))                                  
+                                    (dispatch [:map/grab {:x x :y y}])
+
+                                    ;; wheel button pressed
+                                    (= wheel-button (.-button evt))
+                                    (dispatch [:map/zoom-rectangle-grab {:x x :y y}]))))
+               
+               :on-mouse-move (fn [evt]
+                                (let [x (-> evt .-nativeEvent .-offsetX)
+                                      y (-> evt .-nativeEvent .-offsetY)]
+                                  (cond
+                                    grab
+                                    (dispatch [:map/drag {:x x :y y}])
+
+                                    zoom-rectangle
+                                    (dispatch [:map/zoom-rectangle-update {:x x :y y}]))))
+               :on-mouse-up (fn [evt]
+                              (.stopPropagation evt)
+                              (.preventDefault evt)
+                              (cond
+                                (= left-button (.-button evt))
+                                (dispatch [:map/grab-release])
+
+                                (= wheel-button (.-button evt))
+                                (dispatch [:map/zoom-rectangle-release])))}
+         
+         ;; Controls
+         [controls {:dec-time-fn dect
+                    :inc-time-fn inct
+                    :time time}]
+
+         ;; SVG data map
+         [:svg {:xmlns "http://www.w3.org/2000/svg"
+                :xmlns:amcharts "http://amcharts.com/ammap"
+                :xmlns:xlink "http://www.w3.org/1999/xlink"
+                :version "1.1"
+                :width "100%"
+                :height "100%"
+                :id "map-and-data"}
+
+          ;; gradients definitions
+          [:defs {}
+           [:linearGradient {:id "grad"}
+            [:stop {:offset "0%" :stop-color :red}]
+            [:stop {:offset "100%" :stop-color :yellow}]]]
+
+          ;; map background
+          [:rect {:x "0" :y "0" :width "100%" :height "100%" :fill (:background-color theme)}]
+          
+          ;; map and data svg
+          [:g {:transform (gstr/format "translate(%f %f) scale(%f %f)"
+                                       (or translate-x 0)
+                                       (or translate-y 0)
+                                       scale scale)}
+           [:svg {:view-box "0 0 360 180"}
+            ;; map group
+            (when geo-json-map
+              [:g {}
+               (binding [svg-renderer/*theme* theme]
+                 (svg-renderer/geojson->svg geo-json-map))])
+
+            ;; data group
+            (when analysis-data
+              [:g {}
```

Comment:
empty map needed?

Reply:
Not needed

Reply:
there's couple of places I saw this, but I marked it here only